modules/file-rotation/file-rotation.c

```diff
@@ -24,11 +24,45 @@
 #include "file-rotation.h"
 #include "modules/affile/file-signals.h"
 #include "driver.h"
+#include "time.h"
 
 static void
 _slot_file_rotation(FileRotationPlugin *self, FileFlushSignalData *data)
 {
   msg_trace("File rotation signal received", evt_tag_str("filename", data->filename));
+
+  g_assert(data);
+
+  if(self->number_of_time_rotated > self->number_of_file_rotations)
+    {
+      msg_error("File rotation limit reached", evt_tag_str("filename", data->filename));
+      return;
+    }
+
+  if(data->size >= self->size)
+    {
+      //time_t now = time(NULL);
+      //struct tm *tm = localtime(&now);
+      //gchar date[101];
+
+      //strftime(date, 100, self->date_format, tm);
+      //gchar *new_filename = g_strconcat(data->filename, data->number_of_rotations, NULL);
+
+      gchar *new_filename = g_strdup_printf("%s.%ld", data->filename, self->number_of_time_rotated);
+
+      msg_info("File size reached the limit, rotating file",
+               evt_tag_str("filename", data->filename),
+               evt_tag_str("new_filename", new_filename));
+
+      rename(data->filename, new_filename);
+
+      g_free(new_filename);
+
+      self->number_of_time_rotated++;
```

Comment:
As I see you've started working on the `rotate()` option. The implementation of this option is not trivial: we need to check already existing files and take actions based on what we found in the directory: rename existing 1, 2, 3-suffixed files so that we can use the "1" prefix for the currently rotated file, delete the file that falls out of the `rotate(N)` range, etc.

(I'm more than happy if you're willing to work on it, but I'd like to mention that it's not a requirement for the project.)

Reply:
Have implemented between [line 38](https://github.com/MrAnno/syslog-ng/blob/581d2c2c43b3f3f851bcba4b89efd1d09d82676b/modules/file-rotation/file-rotation.c#L38) to [line 56](https://github.com/MrAnno/syslog-ng/blob/581d2c2c43b3f3f851bcba4b89efd1d09d82676b/modules/file-rotation/file-rotation.c#L56) kindly have a look

Reply:
I think [this](https://github.com/MrAnno/syslog-ng/blob/51bb5fed0c91897e69ea47912fefbe16df9a482b/modules/file-rotation/file-rotation.c#L59) will take care of that.